app/indexers/allinson_flex/dynamic_indexer_behavior.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+# OVERRIDE AllinsonFlex v0.1.0 to turn URI's into human readable strings
+#
+# For whatever reason our decorator pattern was not overriding #generate_solr_document
+# If you check AllinsonFlex::DynamicIndexerBehavior.ancestors, you get what you expect
+# with a decorator.  If you checked the #source_location of :generate_solr_document
+# you also get what you expect.  However, when running a reindex, I was not going
+# through the decorator.  I decided since this is only one method, overriding the entire
+# mixin was acceptable for now.
+
+module AllinsonFlex
+  module DynamicIndexerBehavior
+    include UriToStringBehavior
+    extend ActiveSupport::Concern
+
+    RANGE = "http://www.w3.org/2001/XMLSchema#anyURI"
+
+    included do
+      class_attribute :model_class
+    end
+
+    def generate_solr_document
+      dynamic_schema_service = object.dynamic_schema_service
+
+      super.tap do |solr_doc|
+        dynamic_schema_service.indexing_properties.each_pair do |prop_name, prop_value|
+          prop_value.each do |index_value|
```

Comment:
Wondering if the call to `object.send(prop_name)` has already happened in `super`?  Put another way, could use use:

```ruby
dynamic_schema_service.indexing_properties.each_pair do |prop_name, prop_value|
  next unless uri_properties(dynamic_schema_service).include?(prop_name.to_s)
  prop_value.each do |index_value|
    solr_doc[index_value] = uri_to_value_for(object.send(prop_name))
  end
end
```

On a quick scan I was assuming that the `prop_name` was the name of the property and the `prop_value` was the values of the record.  However, that doesn't quite make sense.  After a deeper read, I'm assuming that `prop_value` is an array of index field names to use for the given `prop_name`.  If my closer read is correct, consider renaming `prop_value` to `index_field_names` and then the inner loop's `index_value` =>  `index_field_name`.

Reply:
I'm not sure if  `object.send(prop_name)` for specifically what we need.  `prop_value` comes from the m3 profile so I don't think it would be handled in `super`.

```rb
[25, 34] in /app/samvera/hyrax-webapp/app/indexers/allinson_flex/dynamic_indexer_behavior.rb
   25:
   26:       super.tap do |solr_doc|
   27:         dynamic_schema_service.indexing_properties.each_pair do |prop_name, prop_value|
   28:           prop_value.each do |index_value|
   29:             debugger
=> 30:             solr_doc[index_value] = if uri_properties(dynamic_schema_service).include?(prop_name.to_s)
   31:                                       uri_to_value_for(object.send(prop_name))
   32:                                     else
   33:                                       object.send(prop_name)
   34:                                     end
(byebug) prop_name
:witness
(byebug) prop_value
["witness_ssm", "witness_tesim"]
```

these variable names are the ones used by [AllinsonFlex](https://github.com/samvera-labs/allinson_flex/blob/main/app/indexers/allinson_flex/dynamic_indexer_behavior.rb#L14-L24).  Perhaps it should be changed there as well?